fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBMetaDataStore.java

```diff
@@ -179,13 +187,38 @@ public void setEvolutionValidator(@Nonnull MetaDataEvolutionValidator evolutionV
         this.evolutionValidator = evolutionValidator;
     }
 
-    @Nullable
+    /**
+     * Get the extension registry used when deserializing meta-data proto messages.
+     *
+     * @return the extension registry used when parsing meta-data
+     */
+    @Nonnull
     protected ExtensionRegistry getExtensionRegistry() {
         return extensionRegistry;
     }
 
+    /**
+     * Set the extension registry used when deserializing meta-data proto messages. This can be set to the
+     * default value by passing a {@code null} extension registry. By default, the extension registry will
+     * have only the extension options specified in "{@code record_metadata_options.proto}". Note that this
+     * differs from the {@linkplain ExtensionRegistry#getEmptyRegistry() empty extension registry}.
+     *
+     * <p>
+     * When the meta-data proto is built into a {@link RecordMetaData} object, extension options on fields
+     * specifying primary key and index information are ignored. However, options specifying each record's
+     * type is still processed, including information specifying the meta-data's union descriptor. As a result,
+     * if the provided registry does not at least register that extension, the {@code RecordMetaData} may
+     * fail to build.
+     * </p>
+     *
+     * @param extensionRegistry the extension registry used when parsing meta-data
+     */
     public void setExtensionRegistry(@Nullable ExtensionRegistry extensionRegistry) {
-        this.extensionRegistry = extensionRegistry;
+        if (extensionRegistry == null) {
+            this.extensionRegistry = DEFAULT_EXTENSION_REGISTRY;
```

Comment:
When would we need to pass null to this method? This is already the default value of `this.extensionRegistry`. 

I actually think, probably instead, keep `this.extensionRegistry` nullable but with a default value of `DEFAULT_EXTENSION_REGISTRY`. Then users can use this method to unset it by passing null, if they truly need it (note that in past it was nullable any way, so it's not worst than before, but this way, it keeps the flexibility to behave the way it used to and also allows users to get the best out of your new change).

Reply:
I guess the use case I was thinking of was someone who had set it to something else and then wanted to get back to the default value. But maybe that's not that compelling a use case.